### ISSUE TITLE
More items for the to-do list

### ISSUE THREAD

**Issue by dppalomar** [dppalomar/sparseIndexTracking#2]
- Obviously, revise the existing code and vignette.
- Allow for box constraints (not allowed already?)
- Include optional mean and variance terms in the objective apart from the tracking error (like in the risk parity portfolio package)
- Include a turnover constraint
- Group sparsity?

**Comment by niheaven** [comment]
What about industrial neutral?

**Comment by dppalomar** [comment]
Indeed, we should allow general linear equality and inequality constraints.

In fact, this package is long due a serious revision. I think we will do soon… cross fingers…

**Comment by snth** [comment]
That would be great. I've watched a couple of your lectures and gone over the papers and this work is impressive. Also really great that you are providing an open source implementation. So thank you very much!

I came here to see if the box and linear equality constraints were implemented and glad to see that you might do so soon! 🤞

**Comment by dppalomar** [comment]
I plan to get to this in several months. I am writing a book now and I need to finish it first. The book has one chapter on index tracking with some new material and perspective. And I will make sure it is online for free.
